source/tyr/migrations/versions/1c8334cd1997_add_equipment_provider.py

```diff
@@ -0,0 +1,32 @@
+"""
+Add equipments provider
+
+Revision ID: 1c8334cd1997
+Revises: 52d628e535db
+Create Date: 2019-03-26 15:16:17.438767
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = '1c8334cd1997'
+down_revision = '52d628e535db'
+
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+
+def upgrade():
+    op.create_table('equipments_provider',
+    sa.Column('id', sa.Text(), nullable=False),
+    sa.Column('instances', postgresql.ARRAY(sa.Text()), nullable=False),
+    sa.Column('klass', sa.Text(), nullable=False),
+    sa.Column('discarded', sa.Boolean(), nullable=False),
+    sa.Column('args', postgresql.JSONB(), server_default='{}', nullable=True),
+    sa.Column('created_at', sa.DateTime(), nullable=False),
```

Comment:
For all attributs nullable=False, don't we need  "server_default='false' or 'true' "as in https://github.com/CanalTP/navitia/blob/cff21de8d98540ada4a6193f998f8023768d740d/source/sql/alembic/versions/29fc422c56cb_add_areas_in_stop_points.py#L21 ?

Reply:
Indeed, it might be better to have a default value. However, in many migrations files, there are also a lot of attributes with 'nullable=False' and no 'server_default' value

Reply:
In that case the function "def __init__(self, id, json=None) should always send a valid json instead of None as default value. Non?

Reply:
No. If you look at the code, if an empty json is passed, the function will look for keys that are absent. The json has to be valid if passed and it is checked in tyr before adding to db